docs/interaction_based_testing.adoc

```diff
@@ -538,11 +538,12 @@ is only meant for documentation purposes and doesn't carry any semantics.
 === Mocking Classes
 
 Besides interfaces, Spock also supports mocking of classes. Mocking classes works
-just like mocking interfaces; the only additional requirement is to put `cglib-nodep-2.2` or higher
-and `objenesis-1.2` or higher on the class path. If either of these libraries is missing from
-the class path, Spock will gently let you know.
+just like mocking interfaces; the only additional requirement is to put `byte-buddy` or
+`cglib-nodep-2.2` or higher on the class path. When using normal `Mock`s or `Stub`s or `Spy`s that are
+configured accordingly, it is also necessary to put `objenesis-1.2` or higher on the class path.
+If either of these libraries is missing from the class path, Spock will gently let you know.
 
-NOTE: Java 8 is only supported from CGLIB 3.2.0 onwards.
+NOTE: Java 8 is only supported with Byte Buddy or CGLIB from 3.2.0 onwards.
```

Comment:
Since we only support java 8 doesn't it make more sense to just increase the version numbers? Spock builds against cglib 3.2.10 and Objenesis-3.0.1

Reply:
You tell me, I just added Byte Buddy so far :-)

Reply:
The Leonard's remark sounds reasonable. I propose to remove the last `NOTE:` at all (sorry @Vampire to do not spot it earlier) and change cglib and objenesis versions in the main text from:
`cglib-nodep-2.2` or higher
to
`cglib-nodep` 3.2.0+

and
`objenesis-1.2` or higher
to
`objenesis` 3.0+

and
`byte-buddy`
to
`byte-buddy` 1.9+
(I don't know what is the minimal compatible version of byte-buddy)

I doubt we want to update those version every time they are changes in Spock itself.

@leonard84 Would it be ok for you?